### ISSUE TITLE
Conflicting Member Variable m_settings in SocketInitiator and Initiator

### ISSUE THREAD

**Issue by pyy07** [quickfix/quickfix#614]
Hello QuickFIX Community,
I've encountered a potential issue regarding the member variable m_settings in both the SocketInitiator class and its parent class Initiator. It appears that both classes have a member variable with the same name, which could lead to unexpected behavior due to variable shadowing or access control conflicts.

![V4K4V6 %B{GUAMF ZARS669](https://github.com/user-attachments/assets/0faddcd3-f3ab-4e27-b494-7bd479432116)
![C~NNT75{TH4%8X~()PQLB61](https://github.com/user-attachments/assets/9f0d08d4-6004-49ea-96e2-8b4fc9bcb46d)

**Comment by cuongth** [comment]
Yes, in derived class SocketInitiator has another variable 'm_settings'. While in base class Initiator, 'm_settings' is protected so that other derived classes can access it. I've investigated SocketInitiator, this additional (unneeded) variable is not harmful.
But I have to investigate other derived classes: ThreadedSSLSocketInitiator, SSLSocketInitiator, ThreadedSocketInitiator.

**Comment by orenmnero** [comment]
Fixed in commit b013d992 — PR #625 removed the duplicate `m_settings` members from `SocketInitiator` and `ThreadedSocketInitiator` that were shadowing the base class `Initiator::m_settings`.
